tarfile/file.go

```diff
@@ -50,3 +50,19 @@ func ExtractFileFromTar(r io.Reader, filePath string) ([]byte, error) {
 	}
 	return nil, fmt.Errorf("file %s not found in tar", filePath)
 }
+
+func UncompressedLayerSize(r io.Reader) (int64, error) {
```

Comment:
https://pkg.go.dev/github.com/google/go-containerregistry@v0.6.0/pkg/v1/partial#UncompressedSize

Reply:
we cannot use this method. It returns byte lenght. But we have calculate size from headers of tar file.

Reply:
Could you give me examples to ensure this method calculates layer sizes correctly?

```
docker history --human=false --format "{{ .Size }}" python:3.9-alpine
0
8173486
0
0
0
32
29361243
0
0
1761413
0
0
0
5613158
```

If you calculate layer sizes of this image by this function, will the values be the same?

Reply:
https://github.com/tonaim/compressedsize

![image](https://user-images.githubusercontent.com/65420264/128391794-a1afa4b4-0376-4115-ac1a-558b5ee634cc.png)

![image](https://user-images.githubusercontent.com/65420264/128391950-89261ef9-a19e-46a6-8899-1f8aec8b0a24.png)


Please observe the layer sizes are near equal. The diff is also because of symlink or corrupted files.

Reply:
Even if the file is a symlink, it will be included as a tar entry and the size should be equal, right? For example, 1,747,444 and 1,761,891 look a big difference. If they don't match, I wonder if the implementation in fanal is also not good.